scene/2d/tile_map_layer.cpp

```diff
@@ -2111,6 +2111,8 @@ void TileMapLayer::notify_tile_map_change(DirtyFlags p_what) {
 }
 
 void TileMapLayer::internal_update() {
+	MutexLock lock(dirty.mutex);
```

Comment:
This lock might be too strict and cause performance problems, would need to be evaluated, it locks quite a large block of code, might be better to lock more specifically to the critical sections

CC @RandomShaper

Reply:
It basically just prevents set_cell from running during internal_update, which seemed pretty reasonable. ~all of the _ helper functions here can touch the dirty cell list, so it seemed pretty fairly scoped. I'm open to other ideas though.

Reply:
A lot of the stuff in those methods do things unrelated to it though, so unless those are also critical theras potential stuff there

Also unsure if this is the only affected list, or if some other solution is better here, i.e. preventing multiple threads from touching this at all, I'm not even sure how that is even happening here